### PR TITLE
#600 Refactor to Equals in Path String, Adds Regression Tests

### PR DESCRIPTION
**Pull request by ZestyBread**
#600 Fix

### CODE REVIEW

**Comment by ZestyBread** [comment]
@Tratcher @muratg Please review

File: src/Microsoft.AspNetCore.Http.Abstractions/PathString.cs [hunk 1]
```diff
@@ -213,7 +213,7 @@ public string Add(QueryString other)
         /// <returns>True if both PathString values are equal</returns>
         public bool Equals(PathString other)
         {
-            return string.Equals(_value, other._value, StringComparison.OrdinalIgnoreCase);
+            return this.Equals(other, StringComparison.OrdinalIgnoreCase);
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Http.Abstractions/PathString.cs]: Remove `this.`

File: src/Microsoft.AspNetCore.Http.Abstractions/PathString.cs [hunk 2]
```diff
@@ -238,7 +242,7 @@ public override bool Equals(object obj)
             {
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Http.Abstractions/PathString.cs]: PathString.Empty.Equals((object)null) == false where PathString.Empty.Equals((PathString)null) == true seems inconsistent.

Comment (Tratcher) [inline src/Microsoft.AspNetCore.Http.Abstractions/PathString.cs]: Return !HasValue?

File: test/Microsoft.AspNetCore.Http.Abstractions.Tests/PathStringTests.cs [hunk 3]
```diff
@@ -16,6 +16,33 @@ public void CtorThrows_IfPathDoesNotHaveLeadingSlash()
             ExceptionAssert.ThrowsArgument(() => new PathString("hello"), "value", "The path in 'value' must start with '/'.");
         }
 
+        [Fact]
+        public void Equals_EmptyPathStringAndDefaultPathString()
+        {
+            // Act and Assert
+            Assert.Equal(PathString.Empty, default(PathString));
```
Comment (Tratcher) [inline test/Microsoft.AspNetCore.Http.Abstractions.Tests/PathStringTests.cs]: Make sure the different overloads are covered:
Assert.True(PathString.Empty == default(PathString));
Assert.True(PathString.Empty.Equals(default(PathString));

Also add variations with the arguments reversed.

File: test/Microsoft.AspNetCore.Http.Abstractions.Tests/PathStringTests.cs [hunk 4]
```diff
@@ -16,6 +16,38 @@ public void CtorThrows_IfPathDoesNotHaveLeadingSlash()
             ExceptionAssert.ThrowsArgument(() => new PathString("hello"), "value", "The path in 'value' must start with '/'.");
         }
 
+        [Fact]
+        public void Equals_EmptyPathStringAndDefaultPathString()
```
Comment (Tratcher) [inline test/Microsoft.AspNetCore.Http.Abstractions.Tests/PathStringTests.cs]: Add a test for GetHashCode

**Comment by Tratcher** [comment]
:shipit:

**Pull request merged**